docs/concepts/projects/init.md

```diff
@@ -53,188 +58,235 @@ if __name__ == "__main__":
 
-The project structure looks the same as a library:
+As with a [packaged application](#packaged-applications), a `src` layout is used. A `py.typed`
+marker is included to indicate to consumers that types can be read from the library:
 
 ```console
-$ uv init --app --package example-packaged-app
-$ tree example-packaged-app
-example-packaged-app
+$ tree example-lib
+example-lib
 ├── .python-version
 ├── README.md
 ├── pyproject.toml
 └── src
-    └── example_packaged_app
+    └── example_lib
+        ├── py.typed
         └── __init__.py
 ```
 
-But the module defines a CLI function:
+!!! note
 
-```python title="__init__.py"
-def main() -> None:
-    print("Hello from example-packaged-app!")
-```
+    A `src` layout is particularly valuable when developing libraries. It ensures that the library is
+    isolated from any `python` invocations in the project root and that distributed library code is
+    well separated from the rest of the project source.
 
-And the `pyproject.toml` includes a script entrypoint:
+A [build system](./config.md#build-systems) is defined, so the project will be installed into the
+environment:
 
-```toml title="pyproject.toml" hl_lines="9 10"
+```toml title="pyproject.toml" hl_lines="12-14"
 [project]
-name = "example-packaged-app"
+name = "example-lib"
 version = "0.1.0"
 description = "Add your description here"
 readme = "README.md"
 requires-python = ">=3.11"
 dependencies = []
 
-[project.scripts]
-example-packaged-app = "example_packaged_app:main"
-
 [build-system]
 requires = ["hatchling"]
 build-backend = "hatchling.build"
 ```
 
-Which can be executed with `uv run`:
+!!! tip
+
+    You can select a different build backend template by using `--build-backend` with `hatchling`,
+    `flit-core`, `pdm-backend`, `setuptools`, `maturin`, or `scikit-build-core`. An alternative
+    backend is required if you want to create a [library with extension modules](#projects-with-extension-modules).
+
+The created module defines a simple API function:
+
+```python title="__init__.py"
+def hello() -> str:
+    return "Hello from example-lib!"
+```
+
+And you can import and execute it using `uv run`:
 
 ```console
-$ uv run --directory example-packaged-app example-packaged-app
-Hello from example-packaged-app!
+$ uv run --directory example-lib python -c "import example_lib; print(example_lib.hello())"
+Hello from example-lib!
 ```
 
-!!! tip
+## Projects with extension modules
+
+Most Python projects are "pure Python", meaning they do not define modules in other languages like
+C, C++, FORTRAN, or Rust. However, projects with extension modules are often used for performance
+sensitive code.
+
+Creating a project with an extension module requires an choosing an alternative build backend. uv
+supports creating projects with the following build backends that support building extension
+modules:
+
+- [`maturin`](https://www.maturin.rs)
+- [`scikit-build`](https://scikit-build.readthedocs.io/en/latest/)
```

Comment:
I'd add a note here that maturin is for rust and scikit-build for C/C++/Fortran/Cython

Reply:
Hm I have that somewhere in some branch.. will add thanks!